types/simpl-schema/simpl-schema-tests.ts

```diff
@@ -0,0 +1,96 @@
+import SimpleSchema from 'simpl-schema';
+
+const StringSchema = new SimpleSchema({
+    basicString: {
+        type: String
+    },
+    limitedString: {
+        type: String,
+        allowedValues: ['pro', 'con']
+    },
+    regExpString: {
+        type: String,
+        regEx: SimpleSchema.RegEx.Id
+    },
+    createdAt: {
+        type: Date,
+        autoValue: () => new Date(),
+    },
+    title: {
+        type: String,
+        label: "Argument",
+        /* TODO, this does not work: "The containing arrow function captures the global value of 'this' which implicitly has type 'any'."
```

Comment:
Help is appreciated here!

Reply:
If you meant for `this` to be a `SchemaDefinition`, I believe you need to use function syntax, rather than arrow syntax: https://www.typescriptlang.org/docs/handbook/functions.html (see the _Type_ section).  However, `SchemaDefinition` does not appear to have a `value` property, so perhaps you meant something else?

Reply:
Sorry, I'm not familiar with the underlying library, so I don't know quite how it's intended to be used.

I could be misreading, but it looks like `custom?: () => any` is on `SchemaDefinition` but the one in the test is actually on the type of `SchemaDefinition.title` (which doesn't seem to be declared?).

Reply:
The parameter is a hash of `SchemaDefinition`s, see here:

```
interface SimpleSchemaStatic {
   new(schema: {[key: string]: SchemaDefinition} | any[], options?: SimpleSchemaOptions): SimpleSchema;
   new(
     schema: {
       [key: string]: SchemaDefinition
         | BooleanConstructor | StringConstructor | NumberConstructor | DateConstructor
         | ArrayConstructor
         | string | RegExp
         | SimpleSchema
     } | any[],
     options?: SimpleSchemaOptions
   ): SimpleSchema;
```
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/31054/commits/250e3e9de236d4c4d74a8c3ce5f08aff00b13f60#diff-21d2fa2cb025853ec8b9fa257855bc15R70

So `SimpleSchema({title: x})` the `x` should be `SchemaDefinition` and it seems to work for all other `SchemaDefinition` files.

The problem is that the callback function passed to `SchemaDefinition.custom` will be executed in a context (bind? ...?) that set's `this.value` among other parameters, which is not part of `SchemaDefinition` or anything. I think I fixed it now:

```
declare CustomValidationContext {
   value: any;
   ...
}
...
        custom?: (this: CustomValidationContext) => undefined | string | SimpleSchemaValidationError;
```

Works for custom: function() { this.value; } not for arrow functions.